eta/core/datasets.py

```diff
@@ -541,38 +541,56 @@ class LabeledDataset(object):
     `self.add_file()`.) Thus it may desirable to use the
     `self.write_manifest()` method to write the internal state to a manifest
     JSON file on disk, at some point after using these methods.
-
-    Attributes:
-        dataset_index: a `LabeledDatasetIndex` containing the paths of data
-            and labels files in the dataset
-        data_dir: the top level directory for the dataset, which would contain
-            manifest.json files
     '''
 
     _DATA_SUBDIR = "data"
     _LABELS_SUBDIR = "labels"
 
-    def __init__(self, dataset_path):
+    def __init__(self, manifest_path):
         '''Creates a LabeledDataset instance.
 
         Args:
-            dataset_path: the path to the `manifest.json` file for the dataset
+            manifest_path: the path to the `manifest.json` file for the dataset
 
         Raises:
             LabeledDatasetError: if the class reading the dataset is not a
                 subclass of the dataset class recorded in the manifest
         '''
-        self.dataset_index = LabeledDatasetIndex.from_json(dataset_path)
+        self._dataset_index = LabeledDatasetIndex.from_json(manifest_path)
         if not isinstance(self, etau.get_class(self.dataset_index.type)):
             raise LabeledDatasetError(
                 "Tried to read dataset of type '%s', from location '%s', "
                 "but manifest is of type '%s'" % (
-                    etau.get_class_name(self), dataset_path,
+                    etau.get_class_name(self), manifest_path,
                     self.dataset_index.type))
-        self.data_dir = os.path.dirname(dataset_path)
+        self._manifest_path = manifest_path
 
         self._build_index_map()
 
+    @property
+    def dataset_dir(self):
+        '''The top level directory for the dataset, which would contain
+        manifest.json files.
+        '''
+        return os.path.dirname(self._manifest_path)
+
+    @property
+    def data_dir(self):
+        '''Deprecated! Use `dataset_dir` instead.'''
+        logger.warning(
```

Comment:
Please run linters as described here: https://github.com/voxel51/eta/blob/develop/docs/linting_guide.md.

This generates "logging not lazy" warning messages. You're supposed to pass formatting args as separate arguments to loggers, which handle rendering the strings themselves, _only_ if needed.

Also, the rest of the codebase formats strings like `"a string %d format %s"`

Reply:
@brimoor what is your approach to linting? I honestly don't bother running linter as it seems very time consuming and also most of the output isn't useful. (I do enjoy the linting within my editor, which is very useful).

I'm asking because I'd like to conform to the codebase but I also don't want to sink a bunch of time into configuring linters right now...

This was the output from running `pylint`. There's nothing about "logging not lazy". `pycodestyle` didn't output anything:
```
datasets.py:2138:0: C0330: Wrong hanging indentation (remove 4 spaces).
                total_frame_count or metadata.total_frame_count)
            |   ^ (bad-continuation)
datasets.py:2756:0: W1401: Anomalous backslash in string: '\|'. String constant might be missing an r prefix. (anomalous-backslash-in-string)
datasets.py:2756:0: W1401: Anomalous backslash in string: '\|'. String constant might be missing an r prefix. (anomalous-backslash-in-string)
datasets.py:44:0: C0103: Constant name "logger" doesn't conform to '(([A-Z_][A-Z0-9_]*)|(__.*__))$' pattern (invalid-name)
datasets.py:44:9: E1101: Module 'logging' has no 'getLogger' member (no-member)
datasets.py:450:0: C0103: Function name "_compute_stride_from_total_frames" doesn't conform to '[a-z_]([a-z0-9_]{0,30})$' pattern (invalid-name)
datasets.py:516:0: R0205: Class 'LabeledDataset' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
datasets.py:741:12: W0212: Access to a protected member _build_index_map of a client class (protected-access)
datasets.py:768:12: W0715: Exception arguments suggest string formatting might be intended (raising-format-tuple)
datasets.py:760:4: R0201: Method could be a function (no-self-use)
datasets.py:952:8: W0621: Redefining name 'type' from outer scope (line 20) (redefined-outer-name)
datasets.py:1319:8: W0621: Redefining name 'type' from outer scope (line 20) (redefined-outer-name)
datasets.py:516:0: R0904: Too many public methods (32/20) (too-many-public-methods)
datasets.py:1648:23: W0621: Redefining name 'type' from outer scope (line 20) (redefined-outer-name)
datasets.py:1744:8: W0621: Redefining name 'type' from outer scope (line 20) (redefined-outer-name)
datasets.py:1742:4: W0221: Parameters differ from overridden 'from_dict' method (arguments-differ)
datasets.py:1779:4: W0107: Unnecessary pass statement (unnecessary-pass)
datasets.py:1782:0: R0205: Class 'LabeledDatasetBuilder' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
datasets.py:1860:12: W0715: Exception arguments suggest string formatting might be intended (raising-format-tuple)
datasets.py:1869:32: C1801: Do not use `len(SEQUENCE)` to determine if a sequence is empty (len-as-condition)
datasets.py:1878:56: W0212: Access to a protected member _DATA_SUBDIR of a client class (protected-access)
datasets.py:1880:33: W0212: Access to a protected member _LABELS_SUBDIR of a client class (protected-access)
datasets.py:1926:4: W0231: __init__ method from base class 'BaseDataRecord' is not called (super-init-not-called)
datasets.py:1971:4: C0111: Missing method docstring (missing-docstring)
datasets.py:1977:4: C0111: Missing method docstring (missing-docstring)
datasets.py:2112:15: E1003: Bad first argument 'BuilderDataRecord' given to super() (bad-super-call)
datasets.py:2161:4: W0235: Useless super delegation in method '__init__' (useless-super-delegation)
datasets.py:2179:0: R0205: Class 'DatasetTransformer' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
datasets.py:2386:8: C0103: Variable name "A" doesn't conform to '[a-z_]([a-z0-9_]{0,30})$' pattern (invalid-name)
datasets.py:2395:20: C0103: Variable name "A" doesn't conform to '[a-z_]([a-z0-9_]{0,30})$' pattern (invalid-name)
datasets.py:2426:11: C1801: Do not use `len(SEQUENCE)` to determine if a sequence is empty (len-as-condition)
datasets.py:2511:15: C1801: Do not use `len(SEQUENCE)` to determine if a sequence is empty (len-as-condition)
datasets.py:2527:15: C1801: Do not use `len(SEQUENCE)` to determine if a sequence is empty (len-as-condition)
datasets.py:2516:4: R0201: Method could be a function (no-self-use)
datasets.py:2551:15: C1801: Do not use `len(SEQUENCE)` to determine if a sequence is empty (len-as-condition)
datasets.py:2563:12: C0103: Variable name "NO_ID" doesn't conform to '[a-z_]([a-z0-9_]{0,30})$' pattern (invalid-name)
datasets.py:2592:15: C1801: Do not use `len(SEQUENCE)` to determine if a sequence is empty (len-as-condition)
datasets.py:2606:12: C0103: Variable name "NO_ID" doesn't conform to '[a-z_]([a-z0-9_]{0,30})$' pattern (invalid-name)
datasets.py:2628:15: C1801: Do not use `len(SEQUENCE)` to determine if a sequence is empty (len-as-condition)
datasets.py:2600:4: R0201: Method could be a function (no-self-use)
datasets.py:2687:15: C1801: Do not use `len(SEQUENCE)` to determine if a sequence is empty (len-as-condition)
datasets.py:2737:15: C1801: Do not use `len(SEQUENCE)` to determine if a sequence is empty (len-as-condition)
datasets.py:2692:4: R0912: Too many branches (14/12) (too-many-branches)
datasets.py:2755:4: C0103: Argument name "A" doesn't conform to '[a-z_]([a-z0-9_]{0,30})$' pattern (invalid-name)
datasets.py:2795:4: C0103: Argument name "A" doesn't conform to '[a-z_]([a-z0-9_]{0,30})$' pattern (invalid-name)
datasets.py:2820:4: C0103: Argument name "A" doesn't conform to '[a-z_]([a-z0-9_]{0,30})$' pattern (invalid-name)
datasets.py:2834:14: C1801: Do not use `len(SEQUENCE)` to determine if a sequence is empty (len-as-condition)
datasets.py:2839:12: C0103: Variable name "Y" doesn't conform to '[a-z_]([a-z0-9_]{0,30})$' pattern (invalid-name)
datasets.py:2854:4: C0103: Argument name "A" doesn't conform to '[a-z_]([a-z0-9_]{0,30})$' pattern (invalid-name)
datasets.py:2854:4: R0201: Method could be a function (no-self-use)
datasets.py:2940:4: C0103: Argument name "A" doesn't conform to '[a-z_]([a-z0-9_]{0,30})$' pattern (invalid-name)
datasets.py:3253:4: W0107: Unnecessary pass statement (unnecessary-pass)
datasets.py:26:0: C0411: standard import "from collections import defaultdict" should be placed before "import six" (wrong-import-order)
datasets.py:27:0: C0411: standard import "import copy" should be placed before "import six" (wrong-import-order)
datasets.py:28:0: C0411: standard import "import glob" should be placed before "import six" (wrong-import-order)
datasets.py:29:0: C0411: standard import "import logging" should be placed before "import six" (wrong-import-order)
datasets.py:30:0: C0411: standard import "import os" should be placed before "import six" (wrong-import-order)
datasets.py:31:0: C0411: standard import "import random" should be placed before "import six" (wrong-import-order)
datasets.py:32:0: C0411: standard import "import re" should be placed before "import six" (wrong-import-order)
```

Reply:
I run `pylint` on everything and quickly scan through the list of warnings similar to this and fix the ones that are relevant. Doesn't take long at all.

Looking at the above list, I see 25% or so that I would actually fix, if I were an author of that file.

For the others, we would ideally (in order of priority):
- configure the logger so that the messages aren't generated
- add linter-ignore messages on a per-case basis if one is willingly breaking some rule
- establish a list of known warnings that we don't obey